### PR TITLE
Prepare for CI 01 -- add python format checks

### PR DESCRIPTION
**Pull request by hao-fang**
See `Makefile` for the format checks. I'll add `CONTRIBUTING.md` in follow-up PRs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,20 @@
...
+# Create and activate a virtualenv
+virtualenv --python=python3.7 venv
+source venv/bin/activate
+
+# Install Python packages
+pip install -r requirements.txt
+
+# Run the main script
+./run.sh
+````
```
Comment (hao-fang) [inline README.md:20]: changed this to use a virtualenv

File: main.py [hunk 2]
```diff
@@ -1,40 +1,45 @@
-from flask import Flask, render_template, render_template_string, make_response
-from flask import jsonify, send_from_directory, redirect
-from flask_frozen import Freezer
-import pickle, json, yaml, csv
-import os, sys, argparse
+# pylint: disable=global-statement,redefined-outer-name
```
Comment (hao-fang) [inline main.py:1]: This file is formatted by `isort` and `black`.
I didn't change the logic.

File: requirements-dev.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+black==19.10b0
+pylint==2.4.4
+mypy==0.761
+-r scripts/requirements.txt
```
Comment (hao-fang) [inline requirements-dev.txt:4]: for developers they will do `pip install -r requirements-dev.txt` so they can run the format checks locally

File: scripts/embeddings.py [hunk 4]
```diff
@@ -1,23 +1,24 @@
-import csv, argparse
-import transformers, torch
-import gc
+import argparse
```
Comment (hao-fang) [inline scripts/embeddings.py:1]: This file is formatted by `isort` and `black`.
I didn't change the logic.

File: scripts/reduce.py [hunk 5]
```diff
@@ -1,18 +1,17 @@
-import torch
-import numpy
-import csv
-import sklearn.manifold
 import argparse
+import csv
```
Comment (hao-fang) [inline scripts/reduce.py:2]: This file is formatted by `isort` and `black`.
I didn't change the logic.

File: scripts/requirements.txt [hunk 6]
```diff
@@ -1,2 +1,3 @@
 transformers
 sklearn
+torch==1.4.0
```
Comment (hao-fang) [inline scripts/requirements.txt:3]: this is needed in `embeddings.py`.

**Comment by hao-fang** [comment]
@srush it seems that I cannot assign people as a reviewer  & merge the PR b/c I don't have write access.

File: README.md (hunk 1 above)
Comment (hao-fang) [reply README.md:20]: hmm... it seems that this breaks the `index.html` when executing `./run.py`.
I'm not sure how to fix this.
If there is no easy fix, I'll revert the changes here.

Comment (srush) [reply README.md:20]: I'm going to remove virtualenv, you can use it, but it's certainly not required.

**Pull request merged**